misk/src/main/kotlin/misk/web/Http.kt

```diff
@@ -32,3 +32,28 @@ annotation class RequestContentType(vararg val value: String)
 
 @Target(AnnotationTarget.FUNCTION)
 annotation class ResponseContentType(val value: String)
+
+/**
+ * When the service is overloaded Misk will intervene and reject calls by returning "HTTP 503
+ * Service Unavailable". We call this load shedding and it works similarly to flow control in TCP.
+ *
+ * We must not shed calls to status endpoints like health checks as doing so may create cascading
+ * failures: overloaded instances that do not respond to health checks will be killed, and this
+ * further overloads the remaining peers.
+ *
+ * Only put this on endpoints that must never be rejected. Such endpoints must not do RPCs, database
+ * queries or other I/O because unexpected latency there can take down the entire service.
+ */
+@Retention(AnnotationRetention.RUNTIME)
+@Target(AnnotationTarget.FUNCTION)
+annotation class AvailableWhenDegraded
+
+/**
+ * Opt-in to concurrency limits.
+ *
+ * TODO(jwilson): make this the default once we're comfortable with the behavior and remove this
+ *     annotation.
+ */
+@Retention(AnnotationRetention.RUNTIME)
+@Target(AnnotationTarget.FUNCTION)
+annotation class ConcurrencyLimitsOptIn
```

Comment:
Shouldn't this be a config option in the yaml so that you can opt in without code changes?

Reply:
Once this is an opt out default, would this annotation be deleted and implicitly be present on all web actions?

Reply:
Exactly

Reply:
Actually I think what I’d prefer over YAML config is runtime flags like LaunchDarkly. For follow-up though.

Reply:
I agree, being able to turn this on with a flag is much better than having to edit YAML and re-deploy.